### PR TITLE
docs(convention): setup commitlint for conventional commits

### PR DESCRIPTION
**Pull request by leomp12**
# Scope of work

- Setup commit messages linter with [commitlint](https://github.com/conventional-changelog/commitlint) and husky as dev dependency;
- Edit on _Contributing_ section of root `README.md`, about commit messages.

### CODE REVIEW

**Comment by filrak** [comment]
Can we setup our own conventions? They should be project-specific imho

File: package.json [hunk 1]
```diff
@@ -55,5 +58,10 @@
       "vue-cli-service lint",
       "git add"
     ]
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
```
Comment (filrak) [inline package.json]: can you also add `yarn test:unit` to husky setup?

**Comment by leomp12** [comment]
> Can we setup our own conventions? They should be project-specific imho so we can have rule per component + general ones like config, docs etc

Yes, we can configure that, in default config the enabled types are:
`feat:`, `fix:`, `chore:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`
we should also document common scopes, eg.:
`feat(component):`, `fix(unit-tests)`, `docs(alpaca):`...
imho it's good for all cases, but if you have better specifications in mind just let me know and I can edit the config file.

File: package.json (hunk 1 above)
Comment (leomp12) [reply package.json]: done

**Comment by filrak** [comment]
If we want to provide meaningful information with each release i guess we should split commits per component so people know immediately where to look for the changes taht they're interested in.

Anyway I think we can leave this PRE open for discussion until we release 1st alpha (hopefully end of april)

**Comment by pspaczek** [comment]
@leomp12 I this issue in #582

**Pull request closed without merging**